decoder/src/log/format.rs

```diff
@@ -21,25 +21,252 @@ pub(super) enum LogSegment {
     Timestamp,
 }
 
-fn parse_argument(input: &str) -> IResult<&str, LogSegment, ()> {
-    let parse_enclosed = delimited(char('{'), take(1u32), char('}'));
-    let mut parse_type = map_res(parse_enclosed, move |s| match s {
-        "f" => Ok(LogSegment::FileName),
-        "F" => Ok(LogSegment::FilePath),
-        "l" => Ok(LogSegment::LineNumber),
-        "s" => Ok(LogSegment::Log),
-        "L" => Ok(LogSegment::LogLevel),
-        "m" => Ok(LogSegment::ModulePath),
-        "t" => Ok(LogSegment::Timestamp),
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub(super) enum LogColor {
+    Black,
+    Red,
+    Green,
+    Yellow,
+    Blue,
+    Magenta,
+    Cyan,
+    White,
+    BrightBlack,
+    BrightRed,
+    BrightGreen,
+    BrightYellow,
+    BrightBlue,
+    BrightMagenta,
+    BrightCyan,
+    BrightWhite,
+
+    /// Color matching the default color for the log level
+    SeverityLevel,
+
+    /// Color matching the default color for the log level,
+    /// but only if the log level is WARN or ERROR
+    WarnError,
+}
```

Comment:
Maybe `LogColor`, could be simplified to following? I didn't think it fully through, but I would be happy if it works out.

```suggestion
pub(super) enum LogColor {
    Color(colored::Color),

    /// Color matching the default color for the log level
    SeverityLevel,

    /// Color matching the default color for the log level,
    /// but only if the log level is WARN or ERROR
    WarnError,
}
```